sql/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java

```diff
@@ -188,7 +187,7 @@ public TransportShardUpsertAction(Settings settings,
                 shardResponse.add(location,
                     new ShardResponse.Failure(
                         item.id(),
-                        userFriendlyMessage(e),
```

Comment:
Would it make sense to change the implementation of `userFriendlyMessage` to use this new logic below?

it seems strange that we are not using a method called `userFriendlyMessage` but add some other logic with a "have nice errors" comment.